src/Processors/QtiV2/In/Interactions/OrderInteractionMapper.php

```diff
@@ -26,7 +26,7 @@ public function getQuestionType()
         /** @var SimpleChoice $simpleChoice */
         foreach ($interaction->getSimpleChoices() as $simpleChoice) {
             $this->orderMapping[$simpleChoice->getIdentifier()] = count($this->orderMapping);
-            $list[] = QtiMarshallerUtil::marshallCollection($simpleChoice->getContent());
+            $list[] = preg_replace('/\s+/', '', QtiMarshallerUtil::marshallCollection($simpleChoice->getContent()));
```

Comment:
@rakeshqdsa this regexp is too greedy, see what it's doing with image tags (just as an example):

```
"list": [
    "<div><imgsrc=\"https://assets.learnosity.com/organisations/561/200507.jpg\"width=\"141\"height=\"143\"/></div>",
    "<div><imgsrc=\"https://assets.learnosity.com/organisations/561/200508.jpg\"width=\"139\"height=\"86\"/></div>",
    "<div><imgsrc=\"https://assets.learnosity.com/organisations/561/200509.jpg\"width=\"159\"height=\"265\"/></div>"
]
```

In the above, we have `imgsrc` (no space) and no space between attributes. Another example is:

```
<div><p><spanclass=\"style-0caccc10f72eb1f266f40f568305cd50\">
```

No space between "span" and "class".

To be honest, you could probably just remove `\n` then `trim()`.

But also, this change looks to only affect OrderInteractions. Is there a base interaction class to do this on all question type stimulus?

Reply:
trim only remove spaces from start and end .
Do you need only remove \n and spaces from start and end of string .

Reply:
Done

Reply:
@rakeshqdsa because these changes are local to OrderInline, I think they should reverted and the changes be added to QtiMarshallerUtil.php, like this:

```
    public static function marshallCollection(QtiComponentCollection $collection)
    {
        $results = [];
        foreach ($collection as $component) {
            if ($component instanceof ObjectElement) {
                $results[] = static::marshallObjectData($component);
            } elseif (!($component instanceof FeedbackInline)) {
                $results[] = static::marshall($component);
            }
        }
        // Clean extra whitespace characters from the returned string
        $cleanedResults = trim(
            str_replace(
                ["\r\n", "\n", "\r", "\t"],
                "",
                implode('', $results)
            )
        );
        return $cleanedResults;
    }
```

This will remove tabs, line breaks, new lines from any string, not just order list.

Reply:
Done